### PR TITLE
Move url collection to WPE plugin

### PR DESCRIPTION
**Pull request by markkelnar**
The collection of urls to the saved mapping is specific to the WPE environment cache.  This change adds a new action and move the collection storage to that [WPE graphql cache plugin](https://github.com/wpengine/wpe-graphql-cache/pull/7).

### CODE REVIEW

File: src/Cache/Collection.php [hunk 1]
```diff
@@ -422,23 +399,7 @@ public function save_query_mapping_cb(
...
-			//phpcs:ignore
-			$url_to_save = wp_unslash( $_SERVER['REQUEST_URI'] );
-
-			// Save the url this query request came in on, so we can purge it later when something changes
-			$urls = $this->store_content( $this->url_key( $request_key ), $url_to_save );
-
-			//phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log, WordPress.PHP.DevelopmentFunctions.error_log_print_r
-			error_log( "Graphql Save Urls: $request_key " . print_r( $urls, 1 ) );
-		}
+		do_action( 'wpgraphql_cache_save_request', $request_key );
```
Comment (jasonbahl) [inline src/Cache/Collection.php:402]: I feel like this action might benefit from passing more params, such as `$query_id`, `$query`, `$variables` and `$operation`.

**Review by jasonbahl — CHANGES_REQUESTED** [review]

File: src/Cache/Collection.php (hunk 1 above)
Comment (markkelnar) [reply src/Cache/Collection.php:402]: Would the `$this->runtime_nodes` and `$this->list_types` be useful in the action from this Collection object?

Comment (jasonbahl) [reply src/Cache/Collection.php:402]: Ya, most likely. I could see decisions needing to be made based of that data.

File: src/Cache/Collection.php [hunk 2]
```diff
@@ -422,30 +399,14 @@ public function save_query_mapping_cb(
...
-			//phpcs:ignore
-			$url_to_save = wp_unslash( $_SERVER['REQUEST_URI'] );
-
-			// Save the url this query request came in on, so we can purge it later when something changes
-			$urls = $this->store_content( $this->url_key( $request_key ), $url_to_save );
-
-			//phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log, WordPress.PHP.DevelopmentFunctions.error_log_print_r
-			error_log( "Graphql Save Urls: $request_key " . print_r( $urls, 1 ) );
-		}
+		do_action( 'wpgraphql_cache_save_request', $request_key, $query_id, $query, $variables, $operation, $this->runtime_nodes, $this->list_types );
```
Comment (markkelnar) [inline src/Cache/Collection.php:402]: @jasonbahl how about this?

Comment (jasonbahl) [reply src/Cache/Collection.php:402]: ❤️

**Review by jasonbahl — APPROVED** [review]

**Pull request merged**